packages/shared_preferences/shared_preferences_ios/test/shared_preferences_ios_test.dart

```diff
@@ -5,113 +5,101 @@
-        }
-        if (methodCall.method == 'remove') {
-          final String key = methodCall.arguments['key'] as String;
-          return await testData.remove(key);
-        }
-        if (methodCall.method == 'clear') {
-          return await testData.clear();
-        }
-        final RegExp setterRegExp = RegExp(r'set(.*)');
-        final Match? match = setterRegExp.matchAsPrefix(methodCall.method);
-        if (match?.groupCount == 1) {
-          final String valueType = match!.group(1)!;
-          final String key = methodCall.arguments['key'] as String;
-          final Object value = methodCall.arguments['value'] as Object;
-          return await testData.setValue(valueType, key, value);
-        }
-        fail('Unexpected method call: ${methodCall.method}');
-      });
-      log.clear();
-    });
-
-    test('registered instance', () {
-      SharedPreferencesIOS.registerWith();
-      expect(
-          SharedPreferencesStorePlatform.instance, isA<SharedPreferencesIOS>());
-    });
-
-    test('getAll', () async {
-      store = SharedPreferencesIOS();
-      testData = InMemorySharedPreferencesStore.withData(kTestValues);
-      expect(await store.getAll(), kTestValues);
-      expect(log.single.method, 'getAll');
-    });
-
-    test('remove', () async {
-      store = SharedPreferencesIOS();
-      testData = InMemorySharedPreferencesStore.withData(kTestValues);
-      expect(await store.remove('flutter.String'), true);
-      expect(await store.remove('flutter.Bool'), true);
-      expect(await store.remove('flutter.Int'), true);
-      expect(await store.remove('flutter.Double'), true);
-      expect(await testData.getAll(), <String, dynamic>{
-        'flutter.StringList': <String>['foo', 'bar'],
-      });
-
-      expect(log, hasLength(4));
-      for (final MethodCall call in log) {
-        expect(call.method, 'remove');
-      }
-    });
-
-    test('setValue', () async {
-      store = SharedPreferencesIOS();
-      expect(await testData.getAll(), isEmpty);
-      for (final String key in kTestValues.keys) {
-        final Object value = kTestValues[key]!;
-        expect(await store.setValue(key.split('.').last, key, value), true);
-      }
-      expect(await testData.getAll(), kTestValues);
-
-      expect(log, hasLength(5));
-      expect(log[0].method, 'setString');
-      expect(log[1].method, 'setBool');
-      expect(log[2].method, 'setInt');
-      expect(log[3].method, 'setDouble');
-      expect(log[4].method, 'setStringList');
-    });
-
-    test('clear', () async {
-      store = SharedPreferencesIOS();
-      testData = InMemorySharedPreferencesStore.withData(kTestValues);
-      expect(await testData.getAll(), isNotEmpty);
-      expect(await store.clear(), true);
-      expect(await testData.getAll(), isEmpty);
-      expect(log.single.method, 'clear');
-    });
+  test('setValueMap', () {
+    expect(() async {
+      await plugin.setValue('flutter.Map', 'key', <String, String>{});
```

Comment:
The `flutter.` prefix should be on the key here, not the type string.

While in practice it doesn't matter since both are unknown values, having it be wrong is needlessly confusing.

Reply:
done